types/v2/format_test.go

```diff
@@ -0,0 +1,154 @@
+package v2
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/grafana/walqueue/types"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDeserializeAndSerialize(t *testing.T) {
+	s := NewFormat()
+	lbls := make(labels.Labels, 0)
+	for i := 0; i < 10; i++ {
+		lbls = append(lbls, labels.Label{
+			Name:  fmt.Sprintf("label_%d", i),
+			Value: randString(),
+		})
+	}
+	for i := 0; i < 100; i++ {
+		aErr := s.AddPrometheusMetric(time.Now().UnixMilli(), rand.Float64(), lbls, nil, nil, nil)
+		require.NoError(t, aErr)
+		aErr = s.AddPrometheusMetadata("name", "unit", "help", "gauge")
+		require.NoError(t, aErr)
+	}
+	kv := make(map[string]string)
+	var bb []byte
+	err := s.Marshal(func(meta map[string]string, buf []byte) error {
+		bb = buf
+		kv = meta
+		return nil
+	})
+	require.NoError(t, err)
+	items, err := s.Unmarshal(kv, bb)
+	require.NoError(t, err)
+	require.Len(t, items, 200)
+	for _, item := range items {
+		ppb := item.Bytes()
```

Comment:
Can we add check that FileFormat() is types.AlloyFileVersionV2 - I saw the PrometheusMetricV1 and wasn't sure what we're testing at first ;)